shared/zkpp.js

```diff
@@ -114,3 +135,26 @@ export const buildRegisterSaltRequest = async (publicKey: string, secretKey: Uin
 
   return [contractSalt, base64ToBase64url(Buffer.concat([nonce, encryptedHashedPasswordBuf]).toString('base64'))]
 }
+
+export const buildUpdateSaltRequestEa = async (password: string, c: Uint8Array): Promise<[string, string]> => {
+  // Derive S_A and S_C as follows:
+  //   -> S_T -< BASE64(SHA-512(SHA-512(T) + SHA-512(c))[0..18]) with T being
+  //     `AUTHSALT` or `CONTRACTSALT`
+  // This way, we ensure both the server and the client contribute to the
+  //   salts' entropy.
+  // When sending the encrypted data, the encrypted information would be
+  // `hashedPassword`, which needs to be verified server-side to verify
+  // it matches p and would be used to derive S_A and S_C.
+  const authSalt = Buffer.from(hashStringArray('AUTHSALT', c)).slice(0, 18).toString('base64')
+  const contractSalt = Buffer.from(hashStringArray('CONTRACTSALT', c)).slice(0, 18).toString('base64')
```

Comment:
Since `'AUTHSALT'` and `'CONTRACTSALT'` string literals are used twice in this file, does it make sense to define them as constants at the top of the file to avoid the risk of typos and to make it clear that these values must be the same in both places?